src/InfoTable.js

```diff
@@ -31,14 +32,30 @@ class InfoTable extends React.Component {
 		);
 	}
 
+	downloadJsonData = () => {
+		if (this.state.noDataText === 'No rows found') {
```

Comment:
`noDataText` can be a boolean .. since the 2 possible outcomes can either be rows found or rows not found.. you can represent this in your state as `true` or `false`

Reply:
There's a third outcome..."Loading" so the user knows that the data is being retrieved from the server, in case of a spotty internet connection.